odxtools/compumethods/scalelinearcompumethod.py

```diff
@@ -1,39 +1,95 @@
+from .compumethod import CompuMethod
+from .linearsegment import LinearSegment
 
 
 @dataclass
 class ScaleLinearCompuMethod(CompuMethod):
-    linear_methods: List[LinearCompuMethod]
+    """A piecewise linear compu method which may feature discontinuities.
+
+    For details, refer to ASAM specification MCD-2 D (ODX), section 7.3.6.6.4.
+    """
 
     @property
-    def category(self) -> CompuMethodCategory:
-        return "SCALE-LINEAR"
-
-    def convert_physical_to_internal(self, physical_value: AtomicOdxType) -> AtomicOdxType:
-        odxassert(
-            self.is_valid_physical_value(physical_value),
-            f"cannot convert the invalid physical value {physical_value!r} "
-            f"of type {type(physical_value)}")
-        lin_method = next(
-            scale for scale in self.linear_methods if scale.is_valid_physical_value(physical_value))
-        return lin_method.convert_physical_to_internal(physical_value)
-
-    def convert_internal_to_physical(self, internal_value: AtomicOdxType) -> AtomicOdxType:
-        lin_method = next(
-            scale for scale in self.linear_methods if scale.is_valid_internal_value(internal_value))
-        return lin_method.convert_internal_to_physical(internal_value)
+    def segments(self) -> List[LinearSegment]:
+        return self._segments
+
+    @staticmethod
+    def compu_method_from_et(et_element: ElementTree.Element, doc_frags: List[OdxDocFragment], *,
+                             internal_type: DataType,
+                             physical_type: DataType) -> "ScaleLinearCompuMethod":
+        cm = CompuMethod.compu_method_from_et(
+            et_element, doc_frags, internal_type=internal_type, physical_type=physical_type)
+        kwargs = dataclass_fields_asdict(cm)
+
+        return ScaleLinearCompuMethod(**kwargs)
+
+    def __post_init__(self) -> None:
+        self._segments: List[LinearSegment] = []
+
+        odxassert(self.category.value == "SCALE-LINEAR",
+                  "ScaleLinearCompuMethod must exibit SCALE-LINEAR category")
+
+        odxassert(self.physical_type in [
+            DataType.A_FLOAT32,
+            DataType.A_FLOAT64,
+            DataType.A_INT32,
+            DataType.A_UINT32,
+        ])
+        odxassert(self.internal_type in [
+            DataType.A_FLOAT32,
+            DataType.A_FLOAT64,
+            DataType.A_INT32,
+            DataType.A_UINT32,
+        ])
+
+        if self.compu_internal_to_phys is None:
+            odxraise("SCALE-LINEAR compu methods require COMPU-INTERNAL-TO-PHYS")
+            return
+
+        compu_scales = self.compu_internal_to_phys.compu_scales
+
+        for scale in compu_scales:
+            self._segments.append(
+                LinearSegment.from_compu_scale(
+                    scale, internal_type=self.internal_type, physical_type=self.physical_type))
+
+    def convert_physical_to_internal(self, physical_value: AtomicOdxType) -> Union[float, int]:
```

Comment:
float as internal value?

Reply:
yeah, the signature of this method compatible with the signature of the base class because it only exhibits a narrower return type. `mypy` is okay with that because the calling code expecting the signature of the base class works with this as well...

Reply:
I am asking if ODX allows internal values to be a float, I though only integers were allowed